cub/test/catch2_test_device_segmented_reduce.cu

```diff
@@ -86,7 +86,7 @@ CUB_TEST("Device reduce works with all device interfaces",
   using params   = params_t<TestType>;
   using item_t   = typename params::item_t;
   using output_t = typename params::output_t;
-  using offset_t = uint32_t;
```

Comment:
important: per discussion on `ChooseOffsetT`, let's make offset type a part of the testing space and add `{u,}int{32,64}` offset types to the coverage.